### PR TITLE
Fix #1659. Replace config empty strings with null

### PR DESCRIPTION
**Pull request by BorisOsipov**
## Proposed changes
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Checklist
- [ ] Checkstyle and unit tests are passed locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by vinogradoff — COMMENTED** [review]
please comment the "getProperties" method to explain this trickery, or rename/refactored to make the purpose clear. The rest looks good for me.

File: src/main/java/com/codeborne/selenide/SelenideConfig.java [hunk 1]
```diff
@@ -345,6 +345,14 @@ public SelenideConfig browserCapabilities(MutableCapabilities browserCapabilitie
   }
 
   private String getProperty(String key, String defaultValue) {
-    return properties.getProperty(key, defaultValue);
+    String value = properties.getProperty(key, defaultValue);
+    return replaceEmptyStringWithNull(value, defaultValue);
+  }
+
+  private String replaceEmptyStringWithNull(String value, String defaultValue) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/SelenideConfig.java:352]: I find this method hard to understand.  :(

It's strange that we do ignore `defaultValue` in most cases, and use it only if it's `null`. 

Probably it would be easier to inline method `replaceEmptyStringWithNull`?

File: src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ static InetSocketAddress getProxyAddress(Proxy proxy) {
   @CheckReturnValue
   @Nonnull
   private Proxy createSeleniumProxy() {
-    return isEmpty(config.proxyHost())
+    return config.proxyHost() == null
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java:146]: I don't see  why this change is useful. 
if `config.proxyHost()` is empty string, the following line will throw an exception anyway.

File: src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public MutableCapabilities createCapabilities(Config config, Browser browser,
 
     ChromeOptions options = new ChromeOptions();
     options.setHeadless(config.headless());
-    if (!config.browserBinary().isEmpty()) {
+    if (config.browserBinary() != null) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java:60]: I don't see why this change is useful.
if config.browserBinary() is empty string, the following line will not work anyway.

File: src/main/java/com/codeborne/selenide/webdriver/InternetExplorerDriverFactory.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public InternetExplorerOptions createCapabilities(Config config, Browser browser
                                                     @Nullable Proxy proxy, @Nullable File browserDownloadsFolder) {
     Capabilities capabilities = createCommonCapabilities(new InternetExplorerOptions(), config, browser, proxy);
     InternetExplorerOptions options = new InternetExplorerOptions(capabilities);
-    if (!config.browserBinary().isEmpty()) {
+    if (config.browserBinary() == null) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/webdriver/InternetExplorerDriverFactory.java:46]: Bug detected! :)

File: src/main/java/com/codeborne/selenide/webdriver/SafariDriverFactory.java [hunk 5]
```diff
@@ -43,7 +43,7 @@ public SafariOptions createCapabilities(Config config, Browser browser,
     if (config.headless()) {
       throw new InvalidArgumentException("headless browser not supported in Safari. Set headless property to false.");
     }
-    if (!config.browserBinary().isEmpty()) {
+    if (config.browserBinary() == null) {
```
Comment (asolntsev) [inline src/main/java/com/codeborne/selenide/webdriver/SafariDriverFactory.java:46]: bug detected here :)

**Review by asolntsev — CHANGES_REQUESTED** [review]

File: src/main/java/com/codeborne/selenide/webdriver/SafariDriverFactory.java (hunk 5 above)
Comment (BorisOsipov) [reply src/main/java/com/codeborne/selenide/webdriver/SafariDriverFactory.java]: Fixed added tests.

File: src/main/java/com/codeborne/selenide/webdriver/InternetExplorerDriverFactory.java (hunk 4 above)
Comment (BorisOsipov) [reply src/main/java/com/codeborne/selenide/webdriver/InternetExplorerDriverFactory.java]: Thx! Fixed

File: src/main/java/com/codeborne/selenide/SelenideConfig.java (hunk 1 above)
Comment (BorisOsipov) [reply src/main/java/com/codeborne/selenide/SelenideConfig.java]: Agree it is not obvious. Inlined.

File: src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java (hunk 3 above)
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java:60]: @BorisOsipov ^^^

File: src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java (hunk 2 above)
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java:146]: @BorisOsipov ^^^

**Review by asolntsev — APPROVED** [review]
